src/test/util_tests.cpp

```diff
@@ -70,9 +71,12 @@ BOOST_AUTO_TEST_CASE(util_datadir)
     args.ClearPathCache();
     BOOST_CHECK_EQUAL(dd_norm, args.GetDataDirBase());
 
+#ifndef WIN32
+    // Windows does not consider "datadir/.//" to be a valid directory path.
     args.ForceSetArg("-datadir", fs::PathToString(dd_norm) + "/.//");
     args.ClearPathCache();
     BOOST_CHECK_EQUAL(dd_norm, args.GetDataDirBase());
+#endif
```

Comment:
In commit "refactor: replace boost::filesystem with std::filesystem" (41d7166c8a598b604aad6c6b1d88ad46e23be247)

The way this is written, it's confusing and vague about windows test behavior. Would change this to actually check windows behavior instead of skipping it, and also add more detail to the comment:

```c++
    args.ForceSetArg("-datadir", fs::PathToString(dd_norm) + "/.//");
    args.ClearPathCache();
#ifndef WIN32
    BOOST_CHECK_EQUAL(dd_norm, args.GetDataDirBase());
#else
    // Windows does not consider "datadir/.//" to be a valid directory path, so
    // the check to see if the directory exists returns false on windows, and
    // GetDataDirBase() returns nothing. When this test was originally written
    // GetDataDirBase() function stripped trailing "/" slashes, so the test was
    // the same on all platforms. But as of #20744, trailing slashes are no
    // longer stripped (because it would have complicated the GetDataDirBase
    // implementation), so the test is now platform-specific.
    BOOST_CHECK(args.GetDataDirBase().empty());
#endif
```

I think trying to clarify things here is worth it, because this test is already very confusing (every time I look at it my brain hurts), and most developers are not using windows, so problems that only happen on windows are especially painful to deal with.

(This is also closer to originally suggested fix from https://github.com/bitcoin/bitcoin/pull/20744#pullrequestreview-785640725)

Reply:
I think the goal was to not allow the unit test to do random crap to the root dir `/`, outside their assigned m_path_root

Reply:
re: https://github.com/bitcoin/bitcoin/pull/20744#discussion_r798597671

> I think the goal was to not allow the unit test to do random crap to the root dir `/`, outside their assigned m_path_root

I don't think that's what's happening here. In any case:

- It is good to replace a vague comment with a specific comment.
- The point of the test is to ensure bitcoin has sane behavior if user uses extra trailing "./" characters in a custom datadir on linux. If we care about this behavior on linux, I don't think think we should say anything goes on windows and not check behavior there. Windows is a 30+ year old operating system with stable path semantics. If path semantics somehow do change, it is a good thing if the test fails, and the test has an understandable comment, we decide can what to do about it.

Reply:
In commit "refactor: replace boost::filesystem with std::filesystem" (41d7166c8a598b604aad6c6b1d88ad46e23be247)

I think this change might be able to be reverted actually, because since this test was disabled a `StripRedundantLastElementsOfPath` call was added to `GetDataDir`. It looks like the call was added in the Dec 23 push https://github.com/bitcoin/bitcoin/pull/20744#event-5807965413 https://github.com/bitcoin/bitcoin/compare/f53be20eade794e55470473f03a8d05d57691c87..2c990b05fae7e1bfef2d3f59044c24274812c559#diff-19427b0dd1a791adc728c82e88f267751ba4f1c751e19262cac03cccd2822216R438 without an explanation